src/array.rs

```diff
@@ -41,8 +53,30 @@ impl<T: RuntimeType> Array<T> {
     }
 }
 
-impl<T> Drop for Array<T> {
+impl<T: RuntimeType> std::ops::Deref for Array<T> {
+    type Target = [T];
+
+    fn deref(&self) -> &[T] {
+        if self.data.is_null() {
+            return &[];
+        }
+
```

Comment:
Just for some sanity checking might be useful to have a `debug_assert!` here that `self.len > 0`.

Reply:
Switched to `is_empty` for consistency.